internal/controllers/app_controller.go

```diff
@@ -272,9 +299,20 @@ func checkPodStatus(c client.Client, appName string, depVersion ketchv1.Deployme
 
 	// check if all pods are running for the deployment
 	for _, pod := range podList.Items {
+		// check if pod have voluntarily terminated with a container exit code of 0
+		if pod.Status.Phase == v1.PodSucceeded {
```

Comment:
not sure that we have good coverage for ready conditions. 
if "web" process has PodSucceeded status then how will it respond to user requests?
if "worker" process has PodSucceeded status then is it ok to return nil? "web" process could not be ready yet

Reply:
`PodSucceeded ` is the phase that indicates if the pod has voluntarily terminated with a container exit code of 0. If it is terminated with some other error code then this condition will not match. The above examples seem more like a user-side error.